public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -338,6 +340,14 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
             this.setIsDirty();
           }
         }
+        if (event.payload.changedObject instanceof SceneDataLayers) {
+          this.setIsDirty();
+        }
+        if (event.payload.changedObject instanceof dataLayers.AnnotationsDataLayer) {
+          if (Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, 'query')) {
+            this.setIsDirty();
+          }
+        }
```

Comment:
I just opened a PR changing the approach of detecting changes #81958, I'll add this there since the approach of detecting changes is different to ensure the changes are really made. 

There is a difference there, and it is I only check if it is an instance of `dataLayers.AnnotationsDataLayer` to be able to save if it is enabled or not as the default value.

Do you think it makes sense?

Reply:
@ivanortegaalba I think we need the `query` property verification in `partialUpdate` because the datalayer will update the `data` property with `PanelData` that might be detected as a dashboard change, when it really isn't. We only care to set the scene dirty when actual annotation query modifications appear in annotations edit page, and these modifications we are interested in live in the `query` property

Reply:
There is one use case not covered here. When we enable or disable the annotation, from the toolbar or the settings, it won't be detected as dirty state, so we won't be able to save that as default values for the dashboard, as we do for variable values.

I'd suggest to do something like 
```js
if (event.payload.changedObject instanceof dataLayers.AnnotationsDataLayer) {
          if (!Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, '$data')) {
            this.setIsDirty();
          }
        }
```
I don't know if `$data` is the right prop to contain, but we can just black list the properties that change when the values are resolved. WDYT?